src/frontend/app.ts

```diff
@@ -140,6 +135,35 @@ export default (octokitApp: App): Express => {
 		)
 	);
 
+	app.get(["/session", "/session/*"], (req: Request, res: Response, next:NextFunction) => {
+		if(!req.params[0]) {
+			return next(new Error("Missing redirect url for session.  Needs to be in format `/session/:redirectUrl`"));
+		}
+		return res.render("session.hbs", {
+			title: "Redirecting...",
+			APP_URL: process.env.APP_URL,
+			redirectUrl: new URL(req.params[0], process.env.APP_URL).href,
+			nonce: res.locals.nonce
+		});
+	});
+
+	// Saves the jiraHost cookie to the secure session if available
+	app.use((req: Request, res: Response, next: NextFunction) => {
+		const jiraHost = req.cookies.jiraHost;
+		if (jiraHost) {
+			// Save jirahost to secure session
+			req.session.jiraHost = jiraHost;
+			// delete jirahost from cookies.
+			res.clearCookie("jiraHost");
+		}
+
+		// Only save xdm_e query when on the post install url (iframe url)
+		if (req.path == postInstallUrl && req.method == "GET" && req.query.xdm_e) {
```

Comment:
/nit we could have it as a middleware function for the postInstall endpoint rather than checking it here.

Reply:
We could, but problem with that is you have your session logic scattered across. I'd rather have it in a single location as it makes it easier to modify later on.